text/0000-explicit-tail-calls.md

```diff
@@ -0,0 +1,648 @@
+    - One point that needs to be decided is if TCE should be a feature that needs to be required from all backends or if it can be optional. Currently, the RFC specifies that an ICE should be issued if a backend cannot guarantee that TCE will be performed.
+    - Another point that needs to be decided is if TCE is supported by a backend what exactly should be guaranteed? While the guarantee that there is no stack growth should be necessary, should performance (as in transforming `call` instructions into `jmp`) also be guaranteed? Note that a backend that guarantees performance should do so **always** otherwise the main intent of this RFC seems to be lost.
+    - Migration guidance, it might be interesting to provide a lint that indicates that a trivial transformation from `return` to `become` can be done for function calls where all requisites are already fulfilled. However, this lint might be confusing and noisy. Decide on if this lint or others should be added.
+    - Should a lint be added for functions that are marked to be a tail call or use become. See the discussion [here](https://github.com/rust-lang/rfcs/pull/3407#discussion_r1159822824), as well as, the clippy and rustfmt changes of an initial [implementation](https://github.com/semtexzv/rust/commit/29f430976542011d53e149650f8e6c7221545207#diff-6c8f5168858fed7066e1b6c8badaca8b4a033d0204007b3e3025bf7dd33fffcb) (2022).
+- What parts of the design do you expect to resolve through the implementation of this feature before stabilization?
+    - Are all calling-convention used by Rust available for TCE with the proposed restrictions on function signatures?
+    - Can the restrictions on function signatures be relaxed?
+        - One option for intra-crate direct calls is to automatically pad the arguments during compilation see [here](https://github.com/rust-lang/rfcs/pull/3407#issuecomment-1500620309). Does this have an influence on other calls? How much implementation effort is it?
+    - Can generic functions be supported?
+    - Can async functions be supported? (see [here](https://github.com/rust-lang/rfcs/pull/1888#issuecomment-1186604115) for an initial assessment)
+    - Can closures be supported? (see [here](https://github.com/rust-lang/rfcs/pull/1888#issuecomment-1186604115) for an initial assessment)
+    - Can dynamic function calls be supported?
+    - Can functions outside the current crate be supported, functions from dynamically loaded libraries?
+    - Can functions that abort be supported?
+    - Is there some way to reduce the impact on debugging and other features?
+- What related issues do you consider out of scope for this RFC that could be addressed in the future independently of
+  the solution that comes out of this RFC?
+  - Supporting general tail calls, the current RFC restricts function signatures which can be loosened independently in the future.
+
+# Future possibilities
```

Comment:
I would like to see a future possibility of a special tail-call-specific calling convention that lifts some of the restrictions, as long as both function use this special calling convention (say `rusttail`).

Reply:
Just to make sure this would be similar to the alternative: "Attribute on Function Declaration"?

So the idea would be to allow using `become` for these marked functions, even if the signatures do not match exactly?

Reply:
Actually, I think that alternative is already close to what you describe: https://github.com/phi-go/rfcs/blob/guaranteed-tco/text/0000-explicit-tail-calls.md#attribute-on-function-declaration.

I would add that this alternative could be adopted in the future additionally to using `become`.

Reply:
Added here: https://github.com/rust-lang/rfcs/pull/3407/commits/3010c2e68c568f6c8741814c20034a5a38086c0e

Reply:
Added here: https://github.com/rust-lang/rfcs/pull/3407/commits/3010c2e68c568f6c8741814c20034a5a38086c0e

Reply:
I'm resolving this as I think it's adressed, please reopen if you disagree.